### PR TITLE
New layout

### PR DESCRIPTION
**Pull request by jcable**
Hi Mohamed, please review and merge this into master if you agree. Comment and ask questions as part of the process!!!!

### CODE REVIEW

File: src/Editor/Editor.js [hunk 1]
```diff
@@ -129,476 +94,263 @@ const styles = theme => ({
...
+      title: "Live broadcast segment",
+      duration: window.duration.toISOString(),
+      startTime: startTime,
+      live: true,
+      insertionType: "live",
+      pid: pid,
+      crid: crid,
+      versionPid: versionPid,
+      versionCrid: versionCrid
+    });
```
Comment (Mo-Sheikh) [inline src/Editor/Editor.js]: The system allows us to add the same live item multiple times. We can possibly add a condition that stops that from happening.

File: src/ScheduleDao/ScheduleDao.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            duration: asset.duration,
+            insertionType: asset.insertionType,
+            asset: asset
+          });
+        });
+      }
+      const sched = new ScheduleObject(sid, date);
+      sched.addFixed(schedule);
+      sched.addGaps();
+      sched.sort();
```
Comment (Mo-Sheikh) [inline src/ScheduleDao/ScheduleDao.js]: Will need to implement the sort type. e.g.  sched.sort((a, b) => {
        return moment(a.startTime) - moment(b.startTime);
      });

File: src/Button.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ class Button extends React.Component {
   };
 
   handleClick = () => {
-    this.setState({ disabled: !this.state.disabled });
+    //this.setState({ disabled: !this.state.disabled });
   };
 
   render() {
```
[the remark this answers is not in the document]
Comment (jcable) [reply src/Button.js]: It's only used in your Editor.spec.js file.

File: src/Loop/Loop.js [hunk 4]
```diff
@@ -1,160 +1,130 @@
...
+      case 2:
+        n[0].insertionType = 'loopStart';
+        n[n.length-1].insertionType = 'loopEnd';
+        break;
+      default:
+        n[0].insertionType = 'loopStart';
+        n[n.length-1].insertionType = 'loopEnd';
+        for(let i=1; i<n.length-1; i++)
+          n[i].insertionType = 'midLoop';
+    }
```
[the remark this answers is not in the document]
Comment (jcable) [reply src/Loop/Loop.js]: There is a guard for empty above.

File: src/Editor/Editor.js (hunk 1 above)
Comment (jcable) [reply src/Editor/Editor.js]: Good spot. Ive fixed this.

File: src/ScheduleDao/ScheduleDao.js (hunk 2 above)
Comment (jcable) [reply src/ScheduleDao/ScheduleDao.js]: It's there

    sort() {
        this.items.sort((a, b) => {
            return a.startTime.isAfter(b.startTime)
                ? 1
                : b.startTime.isAfter(a.startTime)
                    ? -1
                    : 0;
        });
    }

**Pull request merged**